### ISSUE TITLE
Application of DQN Syntax

### ISSUE THREAD

**Issue by Bhaney44** [google-research/bert#552]
I successfully executed the dqn_test.py file:

`from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import tensorflow as tf

from tf_agents.networks import network

class DummyNet(network.Network):
    #Dummy Network for Testing
    def __init__(self, unused_observation_spec, action_spec, name=None):
        super(DummyNet, self).__init__(
                unused_observation_spec, state_space=(), name=name)
        action_spec = tf.nest.flatten(action_spec)[0]
        num_action = action_spec.maximum - action_spec.minimum + 1
        self._layers.append(
                tf.keras.layers.Dense(
                        num_actions,
                        kernel_initializer=tf.compat.v1.initializers.constant([[2, 1],
                                                                               [1, 1]]),
                        bias_intializer=tf.compat.v1.initializers.constant([[1], [1]]),
                        dtype=tf.float32))
    
    def call(self, inputs, unused_step_type=None, network_state=()):
        inputs = tf.cast(inputs, tf.float32)
        for layer in self.layers:
            inputs = layer(inputs)
        return inputs, network_state`

However, I now have two issues. First, how do you make use of this model in an environment setting? Generally, the deep q-network is a reinforcement learning algorithm using a neural network as a function approximator. The learning process is iterative, where an agent learns through experience - making predictions about the value of state-action pairs with a neural network. The goal is for the agent to optimize a reward in an environment.  The problem I am having is understanding how the syntax of the code is associated with the dqn and how to actually implement the agent in an environment. Second, how do understand the relationship between the different files. It is difficult to understand how all of these different files work together, how to use a class, and, how to call function. Is there a way to find resources specifically relating to learning about these tasks within the agents environment? Thanks for the help - would appreciate any advice.
